invenio_pidstore/__init__.py

```diff
@@ -22,11 +22,315 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-"""Invenio module that stores and registers persistent identifiers."""
+"""Module that mints, stores, registers and resolves persistent identifiers.
+
+Invenio-PIDStore is a core component of Invenio responsible for managing a
+mapping from persistent identifiers to internal objects. The mapping is used
+to resolve requests for e.g. an external integer record identifier to an
+internal metadata record.
+
+PIDStore consists of:
+
+- `Persistent identifier API` - for working with persistent identifiers (
+  create, reserve, register, delete, and redirect).
+- `Resolver` - given a persistent identifier retrieve the assigned internal
+  object.
+- `Providers` - wrappers around a persistent identifier to provide extra
+  functionality (e.g. interaction with remote services).
+- `Minters` - small functions that are responsible for minting a specific
+  persistent identifier type for a specific internal object type in as
+  automatic way as possible.
+
+Initialization
+--------------
+First create a Flask application (Flask-CLI is not needed for Flask
+version 1.0+):
+
+>>> from flask import Flask
+>>> from flask_cli import FlaskCLI
+>>> app = Flask('myapp')
+>>> app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite://'
+>>> ext_cli = FlaskCLI(app)
+
+You initialize PIDStore like a normal Flask extension, however Invenio-PIDStore
+is dependent on Invenio-DB so you need to initialize both extensions:
+
+>>> from invenio_db import InvenioDB
+>>> from invenio_pidstore import InvenioPIDStore
+>>> ext_db = InvenioDB(app)
+>>> ext_pidstore = InvenioPIDStore(app)
+
+In order for the following examples to work, you need to work within an
+Flask application context so let's push one:
+
+>>> ctx = app.app_context()
+>>> ctx.push()
+
+Also, for the examples to work we need to create the database and tables (note,
+in this example we use an in-memory SQLite database):
+
+>>> from invenio_db import db
+>>> db.create_all()
+
+Persistent identifiers
+----------------------
+Persistent identifiers map an external persistent identifier into an internal
+object.
+
+- A persistent identifier has **zero or one** assigned internal object.
+- An internal object has **one or many** persistent identifiers.
+- PIDStore can only map to internal objects which are **identified by UUIDs**.
+- Persistent identifiers have a **type** (scheme) which is identified by a
+  6 character string.
+- All persistent identifier values are stored as string values, even if they
+  may have a data type such as integer or UUID.
+
+Persistent identifiers have a **state**:
+
+- New: The persistent identifier is new.
+- Reserved: The persistent identifier is reserved.
+- Registered: The persistent identifier is registered.
```

Comment:
Just below it says "The different states might not all be relevant for different types of identifiers."

DOIs you can reserve which amounts to uploading metadata, but not registering the DOI. For other types it can be something else.